mindsdb/interfaces/jobs/scheduler.py

```diff
@@ -0,0 +1,98 @@
+import random
+import time
+
+import torch.multiprocessing as mp
+
+from mindsdb.utilities.config import Config
+from mindsdb.utilities.log import initialize_log
+from mindsdb.utilities import log
+from mindsdb.interfaces.storage import db
+
+from mindsdb.interfaces.jobs.jobs_controller import JobsExecutor
+
+mp_ctx = mp.get_context('spawn')
+
+logger = log.get_log('jobs')
+
+
+def scheduler_monitor(config):
+    while True:
+
+        logger.debug('Scheduler check timetable')
+        try:
+            check_timetable(config)
+        except (SystemExit, KeyboardInterrupt):
+            raise
+        except Exception as e:
+            logger.error(e)
+
+        # different instances should start in not the same time
+        time.sleep(20 + random.randint(1, 10))
+
+
+def check_timetable(config):
+    scheduler = JobsExecutor()
+
+    for record in scheduler.get_next_tasks():
+        execute_sync(record, config)
+
+
+def execute_sync(record, config):
+    logger.info(f'Job execute: {record.name}({record.id})')
+
+    exec_method = config.get('jobs', {}).get('executor', 'local')
+
+    task_process(record.id, exec_method)
+
+
+def execute_async(record, config):
+    # if async is used with training in background then training is not happening
```

Comment:
Should we check this?

For example, when parsing the query if would be invalid to ask for a train/retrain/adjust job if there is no `USING` clause with `join_learn_process` set to True.

Reply:
Makes sense